mangaki/mangaki/management/commands/tokens.py

```diff
@@ -1,24 +1,53 @@
 from django.core.management.base import BaseCommand
+from django.core.mail import send_mail
 from django.conf import settings
 from mangaki.utils.tokens import compute_token
 from django.contrib.auth import get_user_model
 from django.db import connection
+from django.conf import settings
+import os
+import yaml
+from jinja2 import Template
 
 
 class Command(BaseCommand):
     args = ''
     help = 'Generate tokens for a certain salt'
 
     def add_arguments(self, parser):
-        parser.add_argument('username', nargs=1, type=str)
-        parser.add_argument('salt', nargs=1, type=str)
+        parser.add_argument('username', nargs='+', type=str)
+        parser.add_argument('--salt', type=str, required=True)
+        parser.add_argument('--email-template', type=str)
 
     def handle(self, *args, **options):
-        username = options.get('username')[0]
-        salt = options.get('salt')[0]
-        if username == '*':
-            for user in get_user_model().objects.filter(profile__newsletter_ok=True):
-                if user.email:
-                    pass  # Send a mail
+        usernames = options['username']
+        salt = options['salt']
+
+        if options['email_template'] is None:
+            for username in usernames:
+                self.stdout.write(self.style.SUCCESS(
+                    '{} {}'.format(username, compute_token(salt, username))))
+
         else:
-            self.stdout.write(self.style.SUCCESS(compute_token(salt, username)))
+            filename = os.path.join(settings.DATA_DIR, options['email_template'])
+            with open(filename, 'r') as f:
+                newsletter = yaml.safe_load(f)
+            message = Template(newsletter['body'])
+
+            queryset = get_user_model().objects.filter(profile__newsletter_ok=True)
```

Comment:
No because maybe this code will be used for other newsletter, by keeping a token system but putting an unsubscribe link.

And the main answer is: no, because so far people do not know actually what they are opting out from!